app/page/info/tokens.tsx

```diff
@@ -0,0 +1,229 @@
+import { faCaretDown, faXmark } from "@fortawesome/free-solid-svg-icons"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import * as ethers from "ethers"
+import { useState, type ChangeEvent, type FormEvent } from "react"
+
+import { useProviderContext } from "~app/context/provider"
+import { getChainName } from "~app/page/info/index"
+import { useAccount, useAction, useTokens } from "~app/storage"
+import type { BigNumberish, HexString } from "~typing"
+
+export function Tokens() {
+  const tokens = useTokens()
+  const account = useAccount()
+  const explorerUrl = `https://${getChainName(account.chainId)}.etherscan.io/`
+
+  const [isTokenModalOpened, setIsTokenModalOpened] = useState<boolean>(false)
+  const toggleAccountModal = () => setIsTokenModalOpened(!isTokenModalOpened)
+
+  return (
+    <div>
+      <div className="flex-col justify-center items-center h-auto p-3 border-0 rounded-lg text-base">
+        Tokens:
+        {tokens &&
+          tokens.map((token, index) => {
+            return (
+              <div key={index}>
+                <a
+                  href={`${explorerUrl}token/${token.address}`}
+                  target="_blank">
+                  {`${token.symbol}`}
+                </a>
+                {` ${parseFloat(
+                  ethers.formatUnits(
+                    ethers.toBeHex(token.balance),
+                    ethers.toNumber(token.decimals)
+                  )
+                ).toFixed(6)}`}
```

Comment:
`formatUnits` already takes decimals into account, we don't need to fix it again

```suggestion
                {
                  ethers.formatUnits(
                    token.balance,
                    ethers.toNumber(token.decimals)
                  )
                }
```

Reply:
And why not saving token decimals as plain number?

Reply:
> formatUnits already takes decimals into account, we don't need to fix it again

I initially thought displaying 18 decimals is too long.
But I reverted to showing the full decimals, maybe we can consider a better display in the future.

> And why not saving token decimals as plain number?

I initially thought allowing inputs like `18abc` was fine.
But if [we restrict users to input the decimal](https://github.com/pilagod/waallet-extension/pull/105#discussion_r1606375965), the token input can be plain number directly.

Reply:
Restrict input to only number would be simpler.